### DISCUSSION TITLE
GitLab platformAutomerge and rebasing

### DISCUSSION THREAD

**Discussion by ryancausey** [renovatebot/renovate#19504] [Request Help] [answered]
So I have my project set to require a FF merge, so whenever one PR merges the others must be rebased. This happened to a dev dependency PR that was configured to automerge. It seems that Renovate noticed it had to rebase the PR, but it did not re-enable the platform automerge. Is this expected?

![image](https://user-images.githubusercontent.com/2704075/208825243-f6c48fe5-1e70-401b-8306-29426e9ddc5e.png)

**Comment by rarkins** [comment, marked as answer]
I think it's intentional that platformAutomerge is only enabled at MR creation, so that users can _disable_ it without Renovate fighting them and adding back the flag each time.

However if the flag was still on, and then Renovate force pushes, it would be ideal if it can restore the flag in such cases.

**Reply by ryancausey** [reply]
> However if the flag was still on, and then Renovate force pushes, it would be ideal if it can restore the flag in such cases.

I agree and that's the behavior I was hoping for. The GitLab APIs appear to have the information on the current automerge state available.

Should I make a feature request issue?
